File: backend/src/main/java/com/yourorg/appname/service/TripService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.request.TripCreateRequest;
import com.yourorg.appname.dto.request.TripUpdateRequest;
import com.yourorg.appname.dto.response.TripResponse;

import java.util.List;

public interface TripService {
    List<TripResponse> getCuratedTrips(String category);
    List<TripResponse> getUserTrips(String usernameOrEmail);
    TripResponse getTripById(Long id);
    TripResponse createTrip(TripCreateRequest request, String usernameOrEmail);
    TripResponse updateTrip(Long id, TripUpdateRequest request, String usernameOrEmail);
    void deleteTrip(Long id, String usernameOrEmail);
}

File: frontend/src/services/tripService.js
import apiClient from './apiClient';
import { ENDPOINTS } from '../constants/apiEndpoints';

export const tripService = {
  async getCuratedTrips(category) {
    const params = {};
    if (category && category !== 'ALL') params.category = category;
    const response = await apiClient.get(ENDPOINTS.TRIPS.CURATED, { params });
    return response.data.data;
  },

  async getUserTrips() {
    const response = await apiClient.get(ENDPOINTS.TRIPS.MY_TRIPS);
    return response.data.data;
  },

  async getTripById(id) {
    const response = await apiClient.get(ENDPOINTS.TRIPS.BY_ID(id));
    return response.data.data;
  },

  async createTrip(tripData) {
    const response = await apiClient.post(ENDPOINTS.TRIPS.BASE, tripData);
    return response.data.data;
  },

  async updateTrip(id, tripData) {
    const response = await apiClient.put(ENDPOINTS.TRIPS.BY_ID(id), tripData);
    return response.data.data;
  },

  async deleteTrip(id) {
    const response = await apiClient.delete(ENDPOINTS.TRIPS.BY_ID(id));
    return response.data.data;
  },
};

export default tripService;
